modules/telemetry/events/notification-events.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+namespace VIPWorkflow\Modules\Telemetry\Events;
+
+use VIPWorkflow\Modules\Telemetry\Tracker;
+use WP_Post;
+use WP_User;
+
+function record_notification_sent(
+	WP_Post $post,
+	string $subject,
+	WP_User $user,
+	Tracker $tracker
+) {
+	$tracker->record_event( 'notification_sent', [
```

Comment:
I think we can remove some PII here, maybe even all of it:

- `$subject` can contain user display names and email addresses, which I don't think is necessary for us to track.
- `$post` is an entire WP_Post object which seems like a lot. I was curious what that looked like, here's an example output from `json_encode( (array) get_post( <post_id> ) )`:

    ```json
    {
      "ID": 30303,
      "post_author": "1",
      "post_date": "2024-08-27 19:35:04",
      "post_date_gmt": "0000-00-00 00:00:00",
      "post_content": "<!-- wp:paragraph -->\n<p>Inner content here</p>\n<!-- /wp:paragraph -->",
      "post_title": "fresh post",
      "post_excerpt": "",
      "post_status": "last-pendingreview",
      "comment_status": "open",
      "ping_status": "open",
      "post_password": "",
      "post_name": "",
      "to_ping": "",
      "pinged": "",
      "post_modified": "2024-08-27 19:35:04",
      "post_modified_gmt": "2024-08-27 19:35:04",
      "post_content_filtered": "",
      "post_p arent": 0,
      "guid": "http:/workflow.vipdev.lndo.site/?p=30303",
      "menu_order": 0,
      "post_type": "post",
      "post_mime_type": "",
      "comment_count": "0",
      "filter": "raw"
    }
    ```

    There's a lot of PII in here including the entire `post_content`, `post_excerpt`, and potentially others. I think the most we should track would be a post ID.

- `$user`. Here's a `json_encode( (array) get_user_by( 'id', 1 ) )` for a default `vip dev-env` user:

    ```json
    {
        "data": {
            "ID": "1",
            "user_login": "vipgo",
            "user_pass": "$P$BShVqNoDOqa\/ukp4s.xwYO85QDsIx81",
            "user_nicename": "vipgo",
            "user_email": "vip@localhost.local",
            "user_url": "http://workflow.vipdev.lndo.site",
            "user_registered": "2024-07-03 15:25:18",
            "user_activation_key": "",
            "user_status": "0",
            "display_name": "vipgo"
        },
        "ID": 1,
        "caps": {
            "administrator": true,
            "view_query_monitor": true
        },
        "cap_key": "wp_capabilities",
        "roles": [
            "administrator"
        ],
        "allcaps": {
            "switch_themes": true,
            "edit_themes": true,
            "activate_plugins": true,
            "edit_plugins": true,
            "edit_users": true,
            "edit_files": true,
            "manage_options": true,
            "moderate_comments": true,
            "manage_categories": true,
            "manage_links": true,
            "upload_files": true,
            "import": true,
            "unfiltered_html": true,
            "edit_posts": true,
            "edit_others_posts": true,
            "edit_published_posts": true,
            "publish_posts": true,
            "edit_pages": true,
            "read": true,
            "level_10": true,
            "level_9": true,
            "level_8": true,
            "level_7": true,
            "level_6": true,
            "level_5": true,
            "level_4": true,
            "level_3": true,
            "level_2": true,
            "level_1": true,
            "level_0": true,
            "edit_others_pages": true,
            "edit_published_pages": true,
            "publish_pages": true,
            "delete_pages": true,
            "delete_others_pages": true,
            "delete_published_pages": true,
            "delete_posts": true,
            "delete_others_posts": true,
            "delete_published_posts": true,
            "delete_private_posts": true,
            "edit_private_posts": true,
            "read_private_posts": true,
            "delete_private_pages": true,
            "edit_private_pages": true,
            "read_private_pages": true,
            "delete_users": true,
            "create_users": true,
            "unfiltered_upload": true,
            "edit_dashboard": true,
            "update_plugins": true,
            "delete_plugins": true,
            "install_plugins": true,
            "update_themes": true,
            "install_themes": true,
            "update_core": true,
            "list_users": true,
            "remove_users": true,
            "promote_users": true,
            "edit_theme_options": true,
            "delete_themes": true,
            "export": true,
            "administrator": true,
            "view_query_monitor": true
        },
        "filter": null,
        "\u0000WP_User\u0000site_id": 1
    }
    ```

    Clearly that's a lot of excess data, and includes a lot of PII like emails and security-related data like password hashes.

I think we should just send:

```php
[
    'post' => $post->ID
]
```

and that'll be enough for us to understand how much the notification feature is being used.

Reply:
Yes that makes sense. Outside of PII concerns, that's also just _a lot_ of data that isn't really contextually needed.

Reply:
Removed in 1484e37eb986f780babb89c527958022fbe395e2